### PR TITLE
Stabilize most stream method and remove unnecessary macros

### PR DESCRIPTION
**Pull request by k-nasa**
ref: https://github.com/async-rs/async-std/issues/711

### CODE REVIEW

File: src/io/mod.rs [hunk 1]
```diff
@@ -305,13 +305,9 @@ cfg_std! {
...
-    // For use in the print macros.
-    #[doc(hidden)]
-    pub use stdio::{_eprint, _print};
-
-    pub use stderr::{stderr, Stderr};
-    pub use stdin::{stdin, Stdin};
-    pub use stdout::{stdout, Stdout};
+    pub use stderr::{stderr, Stderr, StderrLock};
+    pub use stdin::{stdin, Stdin, StdinLock};
+    pub use stdout::{stdout, Stdout, StdoutLock};
```
Comment (yoshuawuyts) [inline src/io/mod.rs]: We should probably remove these; they're not very useful at the moment.

```suggestion
```

File: src/io/stderr.rs [hunk 2]
```diff
@@ -1,14 +1,12 @@
 use std::pin::Pin;
 use std::sync::Mutex;
```
Comment (yoshuawuyts) [inline src/io/stderr.rs]: This file can probably be deleted.

File: src/io/stdin.rs [hunk 3]
```diff
@@ -1,16 +1,14 @@
 use std::future::Future;
```
Comment (yoshuawuyts) [inline src/io/stdin.rs]: We should probably delete this file.

File: src/io/stdout.rs [hunk 4]
```diff
@@ -1,14 +1,12 @@
 use std::pin::Pin;
```
Comment (yoshuawuyts) [inline src/io/stdout.rs]: We should probably delete this file.

File: src/stream/mod.rs [hunk 5]
```diff
@@ -317,29 +317,32 @@ mod once;
...
     mod pending;
     mod product;
     mod successors;
     mod sum;
+}
+
+cfg_unstable! {
+    mod from_stream;
+    mod into_stream;
+    mod extend;
```
Comment (yoshuawuyts) [inline src/stream/mod.rs]: This is great!

File: src/stream/product.rs [hunk 6]
```diff
@@ -1,5 +1,6 @@
-use core::pin::Pin;
+use alloc::boxed::Box;
```
Comment (yoshuawuyts) [inline src/stream/product.rs]: `Box` should already be available as a global, shouldn't it?

Comment (yoshuawuyts) [reply src/stream/product.rs]: Oh I guess not in `core`. Makes sense then (: 

https://doc.rust-lang.org/core/prelude/v1/index.html

File: src/io/mod.rs (hunk 1 above)
Comment (k-nasa) [reply src/io/mod.rs]: Deleted them

**Pull request merged**